submissions/7SergSmith7/Friends-app/app.js

```diff
@@ -0,0 +1,160 @@
+              </div>
+              <div class="contact__additional__age">
+                <h4 class="additional__age__title">age</h4>
+                <p class="additional__age_item">${dob.age}</p>
+              </div>
+              <div class="contact__additional__location">
+              <h4 class="additional__location__title">location</h4>   
+              <p class="additional__location_item">${location.country}</p>     
+              </div>
+            </div>
+          </div>`;
+}
+
+optionsMenu.addEventListener("click", onOptionsClick);
+searchInput.addEventListener("change", onChangeSearch);
+
+function onChangeSearch() {
+  contacts = contacts.filter((element) => {
+    return element.name.first
+      .toLowerCase()
+      .includes(searchInput.value.toLowerCase());
+  });
+  renderContacts(contacts);
+}
+
+function renderContacts(contacts) {
+  contactsField.innerHTML = contacts
+    .map((contact) => renderContactTemplate(contact))
+    .join("");
+}
+
+function onOptionsClick(e) {
+  const selectedOption = e.target.classList.value;
+
+  switch (selectedOption) {
+    case "name-sort__z-a":
+      sortName(contacts);
+      renderContacts(sortWithFilter(contacts, filter));
+
+      break;
+    case "name-sort__a-z":
+      sortName(contacts);
+      contacts = contacts.reverse();
+      renderContacts(sortWithFilter(contacts, filter));
+
+      break;
+    case "age-sort__more":
+      sortAge(contacts);
+
+      renderContacts(sortWithFilter(contacts, filter));
+
+      break;
+    case "age-sort__less":
+      sortAge(contacts);
+      contacts = contacts.reverse();
+      renderContacts(sortWithFilter(contacts, filter));
+
+      break;
+    case "gender-filter__man":
+      filter = "male";
+      renderContacts(filterGender(contacts, filter));
+
+      break;
+    case "gender-filter__woman":
+      filter = "female";
+      renderContacts(filterGender(contacts, filter));
+      break;
+    case "options__blocks__reset-btn":
+      resetOptions();
+      break;
+    // case "options__search__icon":
+    //   renderContacts(searchName(searchInput));
+    //   break;
+    default:
+      break;
+  }
+}
+
+function resetOptions() {
+  copyData();
+  filter = "";
+  searchInput.value = "";
+  console.log(contacts);
+  renderContacts(contacts);
+}
+
+function sortAge(contact) {
+  contacts = contact.sort((a, b) => a.dob.age - b.dob.age);
+}
+
+function sortName(contact) {
+  contacts = contact.sort(function (a, b) {
+    if (a.name.first > b.name.first) {
+      return -1;
+    }
+    if (a.name.first < b.name.first) {
+      return 1;
+    }
+    return 0;
+  });
+}
+function filterGender(contacts, gender) {
+  return contacts.filter((element) => element.gender == gender);
+}
+
+// function searchName(input) {
+//   if (input.value)
+//     return originalContacts.filter(
+//       (element) => element.name.first == input.value
+//     );
+// }
+
+function sortWithFilter(contacts, gender) {
```

Comment:
This function looks redundant for me. You can combine it with `filterGender` and just not call `sortWithFilter` if you don't have a gender.

Also, naming of this function is confusing. There is no sorting here and you filtering by `gender`. So it is not `sortWithFilter` in general, it is more like `filterByGenderIfYouHaveAGender`.

And you already have function for filtering by gender.

And by wrapping `if ... else` logic and given wrong name you creating unnecessary nested logic in program.

Reply:
I try fix